zenguard/zenguard.py

```diff
@@ -87,6 +80,22 @@ def detect(self, detectors: list[Detector], prompt: str):
             return {"error": str(e)}
 
         return response.json()
+    
+    def detect_and_send_prompt(self, detectors: list[Detector], prompt: str):
+        try:
+            detected_response = self.detect(detectors, prompt)
+            if "error" in detected_response:
+                return detected_response
+            llm_response = self._llm_client.chat.completions.create(
```

Comment:
we should move this into its own file since we will definitely need to add more. For example, streaming.